lib/sqlalchemy/ext/automap.py

```diff
@@ -670,9 +736,29 @@ class to another, for a collection reference.
     return referred_cls.__name__.lower() + "_collection"
 
 
+class GenerateRelationshipType(Protocol):
+    def __call__(
```

Comment:
**mike bayer** (zzzeek) wrote:

build an overload here...

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4271